Add resource requirements to services

MaxCWhitehead opened the pull request:
Draft PR mostly for review

Changes:

    Update Resources type to contain memory/cpu resources/requests
    Add Resources to ServiceValues (BuildValues templating input now may store resources per service)
    Update composeToServiceValues to read service labels to set resources in ServiceValues
    Update GenerateDeploymentTemplate To override resource requirements on container spec if present in ServicValues
        Implemented for both primary container, and secondary (linked) container if relevant (e.g. the php container in nginx-php linked pod)

TODO:

    Rebase changes here off of latest release instead of main

gchaix commented on internal/testdata/complex/docker-compose.resources.yml:
```diff
@@ -0,0 +1,105 @@
+      lagoon.persistent: /app/docroot/sites/default/files/
+      lagoon.persistent.size: 5Gi
+      lagoon.name: nginx-php
+      lagon.resources.requests.cpu: 20m
```
Is this intentionally `lagon` instead of `lagoon`?

MaxCWhitehead replied:
Nope heh - I noticed a failing test locally so that checks out.

MaxCWhitehead commented:
Updated to allow branch override in service labels:

``` 
# Base resources for service
lagoon.resources.limits.cpu: 200m
lagoon.resources.requests.cpu: 100m

# override on branch "main":
lagoon.resources.override-branch.main.requests.cpu: 400m
lagoon.resources.override-branch.main.limits.cpu: 400m
```

This uses `BuildValues.Branch` when looking up override, these should only apply on branch builds and not PR environments. Updated test and this seems to work.

I looked into using environment variables to specify this stuff, and it feels like a lot of data to try and specify both the service and resource requirements to put into one or multiple env vars. Didn't find anything I was very happy with. I think this approach keeps all of the intended configuration nearby and clearly indicates what service it impacts.

Feedback on exactly what the name/format of branch override is welcome if don't like this version.
